pkg/agent/runner.go

```diff
@@ -250,10 +250,36 @@ func (r *Runner) Run(ctx context.Context, logger *zap.Logger, vmInfoUpdated util
 			}
 		}
 	})
-	r.spawnBackgroundWorker(ctx, logger, "get metrics", func(c context.Context, l *zap.Logger) {
-		r.getMetricsLoop(c, l, func(metrics core.Metrics, withLock func()) {
-			ecwc.Updater().UpdateMetrics(metrics, withLock)
-		})
+	r.spawnBackgroundWorker(ctx, logger, "get system metrics", func(c context.Context, l *zap.Logger) {
+		getMetricsLoop(
+			r,
+			c,
+			l,
+			"system",
+			func() *core.SystemMetrics { return new(core.SystemMetrics) },
+			func() bool { return true },
+			r.global.config.Metrics.System,
+			func(metrics *core.SystemMetrics, withLock func()) {
+				ecwc.Updater().UpdateSystemMetrics(*metrics, withLock)
+			},
+		)
+	})
+	r.spawnBackgroundWorker(ctx, logger, "get LFC metrics", func(c context.Context, l *zap.Logger) {
+		getMetricsLoop(
+			r,
+			c,
+			l,
+			"LFC",
+			func() *core.LFCMetrics { return new(core.LFCMetrics) },
+			func() bool {
+				scalingConfig := r.global.config.Scaling.DefaultConfig.WithOverrides(getVmInfo().Config.ScalingConfig)
+				return *scalingConfig.EnableLFCMetrics // guaranteed non-nil as a required field.
+			},
+			r.global.config.Metrics.LFC,
+			func(metrics *core.LFCMetrics, withLock func()) {
+				ecwc.Updater().UpdateLFCMetrics(*metrics, withLock)
+			},
```

Comment:
Thought about this a bit, and I guess creating another type would require passing some dependencies to it, which will make it more verbose. 

So defining a struct and passing callbacks through it, so that they are named, and we don't have to guess what :question:  `func() bool { return true }` :question:  does.

Reply:
Implemented this in 5dc98a95675fcb6573278834a688b123eda89149